### PR TITLE
[lcms] Update to 2.12

### PR DESCRIPTION
**Pull request by c72578**
**Describe the pull request**

- #### What does your PR fix?  
In addition to updating the version to 2.12, the following
modifications were applied to the port:
  - Use the name `lcms2` for the library, as it is called upstream since
  version 2 of Little-CMS.
  Remark: In the past, the library has already been called `lcms2` for
  a while in vcpkg (1e53c60 until 726c111).
  The ports in vcpkg, which currently require the port lcms
  (devil, libraw and opencolorio), use and support lcms2 include/lib.
  - Add generating a `pkg-config` file
  - Add quotes to file commands in portfile.cmake

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by c72578** [comment]
Added `colmap:x64-windows-static=fail` to `scripts\ci.baseline.txt`.
Info from run:
https://dev.azure.com/vcpkg/public/_build/results?buildId=57750&view=results

**`failure logs for x64-windows-static\colmap\install-x64-windows-static-dbg-err.log`**
```console
ninja: error: 'D:/installed/x64-windows-static/debug/lib/openblas.lib', needed by 'src/exe/colmap.exe',
missing and no known rule to make it
```

File: ports/lcms/portfile.cmake [hunk 1]
```diff
@@ -23,6 +23,7 @@ vcpkg_configure_cmake(
 vcpkg_install_cmake()
 vcpkg_copy_pdbs()
 vcpkg_fixup_cmake_targets()
+vcpkg_fixup_pkgconfig()
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
```
Comment (NancyLi1013) [inline ports/lcms/portfile.cmake:27]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
```
The path should be quoted in cmake files. Please see the [doc ](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/cmake-guidelines.md)here.

File: ports/lcms/portfile.cmake [hunk 2]
```diff
@@ -23,6 +23,7 @@ vcpkg_configure_cmake(
 vcpkg_install_cmake()
```
Comment (NancyLi1013) [inline ports/lcms/portfile.cmake:23]: `file(COPY "${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt" DESTINATION "${SOURCE_PATH}")
`

The path here should also be quoted.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
@cenit Do you remember why you renamed `lcms2 `as `lcms `in this PR https://github.com/microsoft/vcpkg/pull/8087?

**Comment by cenit** [comment]
in that monstrous PR there were so many fixes in so many ports that remembering it might be difficult. Maybe the rationale was this: since the project in CMakeLists.txt was `lcms`, the port name `lcms`, and the CMakeLists.txt was exporting targets, the best way was to have lcms as the name of the library too and have , also because in any case CMake targets hide all these things.
There were no regressions at the time, if the library name has to change for other reasons I don't see any problem.
There was no official reference to any name in the custom CMakeLists.txt, so maybe i just overlooked it

**Comment by cenit** [comment]
I am not sure about the colmap ci error baselined.
It should not pertain this PR, am i right?

**Comment by NancyLi1013** [comment]
> I am not sure about the colmap ci error baselined.
> It should not pertain this PR, am i right?

Yes, this should be a port bug for `colmap `. We need to fix it.

File: scripts/ci.baseline.txt [hunk 3]
```diff
@@ -147,6 +147,7 @@ cmcstl2:x86-windows        = skip
 coin:arm64-windows=fail
 coin:arm-uwp=fail
 coin:x64-uwp=fail
+colmap:x64-windows-static=fail
```
Comment (NancyLi1013) [inline scripts/ci.baseline.txt:150]:
```suggestion
```
This is not related with this PR. It might be better to remove this. We'll fix this in another PR.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/lcms/portfile.cmake [hunk 4]
```diff
@@ -23,6 +23,7 @@ vcpkg_configure_cmake(
...
 vcpkg_copy_pdbs()
 vcpkg_fixup_cmake_targets()
+vcpkg_fixup_pkgconfig()
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
 
 file(INSTALL ${SOURCE_PATH}/COPYING DESTINATION ${CURRENT_PACKAGES_DIR}/share/lcms RENAME copyright)
```
[the remark this answers is not in the document]
Comment (c72578) [reply ports/lcms/portfile.cmake]: The double quotes have been added and the variable `${PORT}` is used now

File: ports/lcms/portfile.cmake (hunk 1 above)
Comment (c72578) [reply ports/lcms/portfile.cmake]: The double quotes have also been added here.

File: ports/lcms/portfile.cmake (hunk 2 above)
Comment (c72578) [reply ports/lcms/portfile.cmake:23]: Double quotes have been added.

File: scripts/ci.baseline.txt (hunk 3 above)
Comment (c72578) [reply scripts/ci.baseline.txt]: OK, `colmap:x64-windows-static=fail` has been removed from **`scripts/ci.baseline.txt`**

**Comment by c72578** [comment]
@NancyLi1013 thanks for your review.
The double quotes have been added to the `file` commands in **`portfile.cmake`** and `${PORT}` is used instead of `lcms` when installing the copyright file. The following line has been removed from **`scripts/ci.baseline.txt`**: `colmap:x64-windows-static=fail`

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
For the failure caused by colmap:x64-windows-static, we're trying to fix it now. I will update the status once a PR is coming.

**Comment by c72578** [comment]
A PR concerning the "openblas - colmap" issue is currently in preparation:
https://github.com/microsoft/vcpkg/pull/19608

**Comment by JackBoosY** [comment]
Depends on https://github.com/microsoft/vcpkg/pull/19628.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
Depends on #19628

**Comment by c72578** [comment]
OK, #19628 has been merged in the meantime:

https://github.com/microsoft/vcpkg/commit/620cb9ad5d67b413c1c2d241680ac01e670881f1

**Comment by NancyLi1013** [comment]
The issue has not been fixed via #19628. So we need to wait for another fix.

**Comment by JackBoosY** [comment]
Well, the colmap regression was not fixed.

**Comment by JackBoosY** [comment]
Will be fixed by https://github.com/microsoft/vcpkg/pull/19608.

**Comment by c72578** [comment]
CI is OK now after today's rebase to master

**Comment by BillyONeal** [comment]
Thanks for your contribution!
